### PR TITLE
PDA sorting fixes & tweaks

### PR DESCRIPTION
**Pull request by Lohikar**
changes:
- Fixed the PDA list not being sorted.
- PDAs now show if they have a pen in them or not in their examine text.
- PDAs can be ctrl-shift-clicked to remove their pen.
- Added a proc designed to simplify common mob sanity checks; see comments for documentation.

### CODE REVIEW

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -1162,26 +1182,26 @@ var/global/list/obj/item/device/pda/PDAs = list()
...
+			if (!cartridge)
+				usr << "<span class='notice'>There is no cartridge in the [name].</span>"
+			else
+				var/turf/T = get_turf(src)
+				cartridge.loc = T
+				if (ismob(loc))
+					var/mob/M = loc
+					M.put_in_hands(cartridge)
+				else
+					cartridge.loc = get_turf(src)
```
Comment (skull132) [inline code/game/objects/items/devices/PDA/PDA.dm]: `forceMove` plz.

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 2]
```diff
@@ -1162,26 +1182,26 @@ var/global/list/obj/item/device/pda/PDAs = list()
...
-		usr << "<span class='notice'>You cannot do this while restrained.</span>"
+	switch (use_check(usr, USE_DISALLOW_SILICONS))
+		if (USE_FAIL_DEAD,USE_FAIL_INCAPACITATED)
+			usr << "<span class='notice'>You cannot do this in your current state.</span>"
+		if (USE_SUCCESS)
+			if (!cartridge)
+				usr << "<span class='notice'>There is no cartridge in the [name].</span>"
+			else
+				var/turf/T = get_turf(src)
+				cartridge.loc = T
```
Comment (skull132) [inline code/game/objects/items/devices/PDA/PDA.dm]: `forceMove`.

**Comment by Lohikar** [comment]
Addressed.

File: code/_helpers/unsorted.dm [hunk 3]
```diff
@@ -1198,3 +1198,37 @@ var/list/WALLITEMS = list(
...
+	if (NOT_FLAG(USE_ALLOW_NONLIVING) && !isliving(user))
+		return USE_FAIL_NONLIVING
+
+	if (NOT_FLAG(USE_ALLOW_NON_ADJACENT) && !Adjacent(user))
+		return USE_FAIL_NON_ADJACENT
+
+	if (NOT_FLAG(USE_ALLOW_DEAD) && user.stat == DEAD)
+		return USE_FAIL_DEAD
+
+	if (NOT_FLAG(USE_ALLOW_INCAPACITATED) && (user.incapacitated())
```
Comment (skull132) [inline code/_helpers/unsorted.dm]: Missing closing parepen.

**Pull request merged**